lib/9pfs/9pfs_vnops.c

```diff
@@ -499,25 +509,31 @@ static int uk_9pfs_write(struct vnode *vp, struct uio *uio, int ioflag)
 	if (!uio->uio_resid)
 		return 0;
 
-	iov = uio->uio_iov;
-	while (!iov->iov_len) {
-		uio->uio_iov++;
-		uio->uio_iovcnt--;
-	}
+	while (i < uio->uio_iovcnt) {
+		iov = &uio->uio_iov[i];
+		if (!iov->iov_len) {
+			i++;
+			continue;
+		}
 
-	rc = uk_9p_write(dev, fid, uio->uio_offset,
-			    iov->iov_len, iov->iov_base);
-	if (rc < 0)
-		return -rc;
+		rc = uk_9p_write(dev, fid, uio->uio_offset,
+				iov->iov_len, iov->iov_base);
+		if (unlikely(rc < 0))
+			goto out;
```

Comment:
The file size might have changed already so we need to execute the check before the `out` label. Just set `rc` here and remove the `rc=0` after the loop.
```suggestion
		if (unlikely(bytes < 0)) {
			rc = -(int)bytes;
			break;
		}
```

Reply:
Shouldn't we set `rc` to `(int) bytes` here, since at the end of the function we already negate `rc` with `return -rc`?